bundles/core/src/main/java/com/adobe/cq/wcm/core/components/commons/editor/dialog/childreneditor/Editor.java

```diff
@@ -62,11 +62,13 @@ private void readChildren() {
             container = resolver.getResource(containerPath);
             if (container != null) {
                 ComponentManager componentManager = request.getResourceResolver().adaptTo(ComponentManager.class);
-                for (Resource resource : container.getChildren()) {
-                    if (resource != null) {
-                        Component component = componentManager.getComponentOfResource(resource);
-                        if (component != null) {
-                            items.add(new Item(request, resource));
+                if(componentManager != null){
```

Comment:
is it possible to come up with a much shorter version using {{Optional}}s?

```
items = container.getChildren().stream()
  .map(res -> componentManager.getComponentOfResource(res)
  .filter(Objects::notNull)
  .map(new Item(request,res)
  .collect(Collectors::toList());
```

(you might want to try to include the earlier statements as well). This reduces the number of branches and makes it easier to understand the code.)

Reply:
Hi @joerghoh I will merge this PR for now as is seems @kotisyamala is not going to change it. There is a pending PR (#814) which actually applies the same fix , so i will try to incorporate your suggestions there.